### ISSUE TITLE
Fallback for testing in browser.

### ISSUE THREAD

**Issue by HbIXA** [GillesC/cordova-plugin-nativestorage#9]
log

```
Error: exec proxy not found for :: NativeStorage :: putBoolean
Missing Command Error
```

**Comment by alokrajiv** [comment]
are you referring to the fact that you are not able to run in the browser during development.

**Comment by HbIXA** [comment]
Yes

**Comment by alokrajiv** [comment]
In the browser there is no preference or 'native storage' available. This is the same reason why many plugins don't work in a browser and work only in the device or emulator.  A possible soln would be to have a fallback to localstorage. As of now we don't have a javascript proxy to fall back.

**Comment by HbIXA** [comment]
I think it makes sense to implement at least a cookie, it is for the browser .

**Comment by alokrajiv** [comment]
See this [ngCordova doc](http://ngcordova.com/docs/common-issues/) link which specifies testing apps with plugins in general on a browser is not a good idea.
It is a common issue with cordova and not specifically this plugin. Same goes with plugins such as media, gyro, camera etc.

**Comment by GillesC** [comment]
@alokrajiv I'll implement a localstorage approach if the application is running in the browser.

**Comment by alokrajiv** [comment]
@GillesC Sounds good.

**Comment by GillesC** [comment]
It'll be a simpel approach: test if in browser, if yes save via localstorage.

**Comment by alokrajiv** [comment]
I suggest you could check for availability of cordova or phonegap. if not fallback to localstorage. because checking if running in browser will be difficult.

**Comment by GillesC** [comment]
cordova provides a cordova.platformId parameter whichs results in 'browser' when running in a browser.

**Comment by alokrajiv** [comment]
ok then. But, I though that was for the 'browser' platform only. Anyways, try it. You can see this open issue [link](https://github.com/Wizcorp/phonegap-facebook-plugin/issues/794). Because usually in the browser cordova is undefined.

**Comment by GillesC** [comment]
Thank you, I was just checking the platformID, now I will check for cordova and phonegap also ;)

**Comment by GillesC** [comment]
saving and stoing when running in a browser will result in invocations on these methods.
I will now begin testing this implementation
```javascript
function inBrowser() {
    return (window.cordova && window.cordova.platformId === 'browser') || !window.phonegap || !window.cordova;
}

/* Method for storage in localstorage if run in browser */
NativeStorage.prototype.setInLocalStorage = function (reference, variable, error) {
    try {
        var varAsString = JSON.stringify(variable);
        localStorage.setItem(reference, varAsString);
    } catch (err) {
        error(err);
    }

};

NativeStorage.prototype.getFromLocalStorage = function (reference, success, error) {
    try {
        var obj = JSON.parse(localStorage.getItem(reference));
        success(obj);
    } catch (err) {
        error(err);
    }
};
```

**Comment by GillesC** [comment]
Testing in browser now possible. Check if your problem is resolved.

Added browser for testing purposes in commit 29b4bb6348732b6b7ca260894f15eaf592e5d0a4.

**Comment by alokrajiv** [comment]
@GillesC are you able to reproduce the results on browser?

**Comment by GillesC** [comment]
Yup I've tested through the browser platform.

**Comment by alokrajiv** [comment]
Working perfectly. I am closing the issue.

**Comment by alokrajiv** [comment]
commit 29b4bb6348732b6b7ca260894f15eaf592e5d0a4.

**Comment by GillesC** [comment]
Yeah, I thought waiting for @HbIXA to respond. And if his issue had resolved, than to close this issue.

**Comment by alokrajiv** [comment]
Ok, lets wait for @HbIXA  to confirm. If he doesnt soon, I'll reopen the issue.

**Comment by alokrajiv** [comment]
Temporarily reopening the issue for 48hrs. @HbIXA pls confirm that your issue is resolved. If not in 48hrs we'll close the issue because we have tested and aren't facing the issue anymore.

**Comment by HbIXA** [comment]
1. rm plugin
2. add plugin 

result : plugin ls
`cordova-plugin-nativestorage 1.0.9 "NativeStorage"`

version not updated

**Comment by alokrajiv** [comment]
`cordova plugin remove cordova-plugin-nativestorage && cordova plugin add https://github.com/GillesC/cordova-plugin-nativestorage.git && cordova plugin ls `

I suspect you didn't install the development version. Bcoz the v1.0.10 isn't on release yet.

**Comment by GillesC** [comment]
Like I said in a previous comment, you must install dthe development version. this because I'm waiting for your response to republish the plugin on npm.

Testing in browser is now possible. Would you check if your problem has been resolved?

Please remove the old version first:
```
cordova plugin remove cordova-plugin-nativestorage
```
And use the development version of the plugin:
```
cordova plugin add https://github.com/GillesC/cordova-plugin-nativestorage.git
```

**Comment by HbIXA** [comment]
good job

**Comment by GillesC** [comment]
I think by 'good job' you meant, that the plugin works ;)
I'll close this issue now.

Added browser for testing purposes in commit 29b4bb6.

**Comment by jayj** [comment]
Thanks for this. I have an issue where the `inBrowser` function returns true for me on Android because `window.phonegap` is undefined.

**Comment by alokrajiv** [comment]
@jayj I understood. But, let me recreate the issue. and I will respond back.

**Comment by alokrajiv** [comment]
@jayj Thank you for pointing , the commit actually broke the functionality. I am re-opening the issue while I update the inBrowser function.

**Comment by alokrajiv** [comment]
I have updated it. Please reinstall the development version and check.
`cordova plugin remove cordova-plugin-nativestorage && cordova plugin add https://github.com/GillesC/cordova-plugin-nativestorage.git`
And retry your use case. I beleive I have corrected the problem with commit 3714c9b319b650d70211af8e5fb7d26c314b9685.
@jayj Please confirm.

**Comment by jayj** [comment]
@alokrajiv thanks for the quick fix! It works now 👍

**Comment by GillesC** [comment]
Glad thinks worked out! I'm closing the issue
Problem corrected with commit 3714c9b
